### PR TITLE
Fix GabutScans Broken Images

### LINKED ISSUE #12257: Gabut Scans: Broken
### Source information

Gabut Scans 1.3.13

### Steps to reproduce

1. Choose any comic in the Gabut Scans extension.
2. Choose a chapter (especially the new chapter),
and it happens "image could not be found". 
3. Almost all comics there (Gabut Scans) have this problem, I've tried in webview but there Normal, nothing happens
photo_2022-06-20_18-41-37
photo_2022-06-20_18-41-35 (2)
photo_2022-06-20_18-41-35

### Expected behavior

Normal as usual

### Actual behavior

"Image could not be found" Almost in all chapters, especially the new chapter

### Tachiyomi version

0.13.4

### Android version

Android 11

### Other details

photo_2022-06-20_18-41-35 (2)

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.13.4](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by mxsgx**
Closes #12257

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

**Comment by FathiRaihan** [comment]
udah beres kah? pas buka di app kok masih sama aja

**Comment by mxsgx** [comment]
> udah beres kah? pas buka di app kok masih sama aja

belum dimerge

**Comment by FathiRaihan** [comment]
> > udah beres kah? pas buka di app kok masih sama aja
> 
> belum dimerge

woke lah, kukira sudah beres. maklum baru pertama kali report issue mwehehehe

File: multisrc/overrides/wpmangareader/gabutscans/src/GabutScans.kt [hunk 1]
```diff
@@ -10,4 +15,33 @@ class GabutScans : WPMangaReader(
...
+
+    override fun pageListParse(document: Document): List<Page> {
+        val pages = mutableListOf<Page>()
+
+        // Prefer using sources loaded from javascript
+        // because current CDN (Statically) can't load old images.
+        // Example: https://gabutscans.com/cinderella-wa-sagasanai-chapter-41-end-bahasa-indonesia/
+
+        val docString = document.toString()
+        val imageListRegex = Regex("\\\"images.*?:.*?(\\[.*?\\])")
```
Comment (stevenyomi) [inline multisrc/overrides/wpmangareader/gabutscans/src/GabutScans.kt:36]:
```suggestion
        val imageListRegex = Regex("""\"images.*?:.*?(\[.*?])""")
```
Better use [raw strings](https://kotlinlang.org/docs/basic-types.html#string-literals). `]` is enought for the right bracket - you will be prompted about this if you use Android Studio.

File: multisrc/overrides/wpmangareader/gabutscans/src/GabutScans.kt [hunk 2]
```diff
@@ -10,4 +15,33 @@ class GabutScans : WPMangaReader(
...
+        val imageListJson = imageListRegex.find(docString)!!.destructured.toList()[0]
```
Comment (stevenyomi) [inline multisrc/overrides/wpmangareader/gabutscans/src/GabutScans.kt:37]:
```suggestion
        val imageListJson = imageListRegex.find(docString)!!.groupValues[1]
```
or at least:
```suggestion
        val (imageListJson) = imageListRegex.find(docString)!!.destructured
```

File: multisrc/overrides/wpmangareader/gabutscans/src/GabutScans.kt [hunk 3]
```diff
@@ -10,4 +15,33 @@ class GabutScans : WPMangaReader(
...
+
+        val imageList = json.parseToJsonElement(imageListJson).jsonArray
+
+        pages += imageList.mapIndexed { i, jsonEl ->
```
Comment (stevenyomi) [inline multisrc/overrides/wpmangareader/gabutscans/src/GabutScans.kt:41]:
```suggestion
        return imageList.mapIndexed { i, jsonEl ->
```

**Review by stevenyomi — CHANGES_REQUESTED** [review]

File: multisrc/overrides/wpmangareader/gabutscans/src/GabutScans.kt (hunk 1 above)
Comment (mxsgx) [reply multisrc/overrides/wpmangareader/gabutscans/src/GabutScans.kt:36]: I just copied it from the parent class 😅 but it's OK 👍

**Pull request merged**